src/main/kotlin/util/UtilFunctions.kt

```diff
@@ -86,4 +100,19 @@ object UtilFunctions {
         return Di[Di.size - 1]
     }
 
+    fun runInThreadPoolWithExceptionHandle(threadPool: ExecutorService, tasks: List<Runnable>) {
```

Comment:
Please consider Kotlin coroutines as an alternative to java.util.concurrent tooling. Coroutines are much easier to use after some getting used to

Reply:
I'll try to use them in future versions.